tests/test_types.py

```diff
@@ -919,6 +919,76 @@ class Model(BaseModel):
         assert Model(foo=value).foo == result
 
 
+@pytest.mark.parametrize('value,result', (
+    ('nonexistentfile', [
+        {
+            'loc': ('foo',),
+            'msg': 'file or directory at path "nonexistentfile" does not exist',
+            'type': 'value_error.path.not_exists',
+            'ctx': {
+                'path': 'nonexistentfile',
+            },
+        },
+    ]),
+    ('tests', [
+        {
+            'loc': ('foo',),
+            'msg': 'path "tests" does not point to a file',
+            'type': 'value_error.path.not_a_file',
+            'ctx': {
+                'path': 'tests',
+            },
+        },
+    ]),
+    ('tests/test_types.py', Path('tests/test_types.py')),
+))
+def test_file_path_validation(value, result):
+    class Model(BaseModel):
+        foo: FilePath
+
+    if not isinstance(result, Path):
+        with pytest.raises(ValidationError) as exc_info:
```

Comment:
Please separate the two cases into different years.

Reply:
Sorry can't understand what you want (different years?), can you explain more detailed?

Reply:
Sorry, I'm on holiday - typo from my phone. 

Separate into different tests.

Reply:
Updated, please review